cmd/lotus-storage-miner/sectors.go

```diff
@@ -440,69 +444,155 @@ var sectorsExtendCmd = &cli.Command{
 		defer nCloser()
 
 		ctx := lcli.ReqContext(cctx)
-		if !cctx.Args().Present() {
-			return xerrors.Errorf("must pass at least one sector number")
-		}
 
-		maddr, err := nodeApi.ActorAddress(ctx)
+		maddr, err := getActorAddress(ctx, cctx)
 		if err != nil {
-			return xerrors.Errorf("getting miner actor address: %w", err)
+			return err
 		}
 
-		sectors := map[miner.SectorLocation][]uint64{}
+		var params []miner3.ExtendSectorExpirationParams
 
-		for i, s := range cctx.Args().Slice() {
-			id, err := strconv.ParseUint(s, 10, 64)
-			if err != nil {
-				return xerrors.Errorf("could not parse sector %d: %w", i, err)
+		if cctx.Bool("v1-sectors") {
+			extensions := map[miner.SectorLocation]map[abi.ChainEpoch][]uint64{}
+			// are given durations within a week?
+			closeEnough := func(a, b abi.ChainEpoch) bool {
+				diff := a - b
+				if diff < 0 {
+					diff = b - a
+				}
+
+				return diff <= 7*builtin.EpochsInDay
 			}
 
-			p, err := api.StateSectorPartition(ctx, maddr, abi.SectorNumber(id), types.EmptyTSK)
+			sis, err := api.StateMinerActiveSectors(ctx, maddr, types.EmptyTSK)
 			if err != nil {
-				return xerrors.Errorf("getting sector location for sector %d: %w", id, err)
+				return xerrors.Errorf("getting miner sector infos: %w", err)
+			}
+
+			for _, si := range sis {
+				if si.SealProof < abi.RegisteredSealProof_StackedDrg2KiBV1_1 {
+
+					ml := builtin3.SealProofPoliciesV11[si.SealProof].SectorMaxLifetime
+					// if the sector's missing less than a week of its maximum possible lifetime, don't bother extending it
+					if closeEnough(si.Expiration-si.Activation, ml) {
+						continue
+					}
+
+					newExp := ml - (miner3.WPoStProvingPeriod * 2) + si.Activation
+					p, err := api.StateSectorPartition(ctx, maddr, si.SectorNumber, types.EmptyTSK)
```

Comment:
This is really inefficient (searches all partitions/deadlines). OK for a first pass, but this could be _very_ slow for some miners.

On the other hand, may be not... let's wait and see.

Reply:
Yeah, I tried it for the largest miners, and it is...not fast. Easy optimization is to only load the actor state once, but that is not gonna be the critical path. Alternative is to iterate over deadlines, and pre-compute one very large map of sectors -> locations, and then hit that up as needed.

This is a "needs to be run once" command, though, so eh?